include/async/GenericMuxedInterruptDistributor.h

```diff
@@ -1,20 +1,107 @@
 // SPDX-FileCopyrightText: Deutsches Elektronen-Synchrotron DESY, MSK, ChimeraTK Project <chimeratk-support@desy.de>
 // SPDX-License-Identifier: LGPL-3.0-or-later
 #pragma once
+#include "../NDRegisterAccessor.h"
+#include "../RegisterPath.h"
 #include "MuxedInterruptDistributor.h"
+#include <nlohmann/json.hpp>
+
+#include <fstream>
+
+using json = nlohmann::json;
 
 namespace ChimeraTK::async {
+  namespace genIntC {
+
+    struct jsonDescriptorStandardV1 {
+      static constexpr int jsonDescriptorVersion = 1;
+      inline static const std::string VERSION_JSON_KEY = "version";
+      inline static const std::string OPTIONS_JSON_KEY = "options";
+      inline static const std::string PATH_JSON_KEY = "path";
+    };
+
+    enum optionCode {
+      ISR = 0,           // Interrupt Status Register
+      IER,               // Interrupt Enable Register
+      MER,               // Master Interrupt Register,
+      MIE,               // Master Interrupt Enable  Functionally equivalent to MER
+      GIE,               // Global Interrupt Enable, Functionally equivalent to MER
+      ICR,               // Interrupt Clear Register
+      IAR,               // Interrupt Acknowledge Register
+      IPR,               // Interrupt Pending Register =ISR & IER, a convenience feature for software.
+      SIE,               // Set Interrupt Enable
+      CIE,               // Clear Interrupt Enable
+      IMaskR,            // Interrupt Mask Register. IMR Acronym Collision. Temporarily not allowed. See
+      IModeR,            // Interrupt Mode Register. IMR Acronym Collision, defined in the standard but not allowed
+      IVR,               // defined in the standard but not allowed
+      ILR,               // defined in the standard but not allowed
+      IVAR,              // defined in the standard but not allowed
+      IVEAR,             // defined in the standard but not allowed
+      OPTION_CODE_COUNT, // used for counting how many valid enums there are here & setting list lengths
```

Comment:
OPTION_CODE_COUNT is necessarily a compile time constant, since it determines the size of the bitset OptionRegisterSettings. Similarly, other OptionCodes are intended to be compile time constants. Therefore, shouldn't the compile time constant naming convention be used?

Reply:
Listed in redmine ticket #13126

Reply:
Decision: keep as-is.